api/src/types/field.ts

```diff
@@ -42,3 +44,12 @@ export type Field = {
 	schema: Column | null;
 	meta: FieldMeta | null;
 };
+
+export type Condition = {
+	name: string;
+	rule: Record<string, any>;
+
+	readonly?: boolean;
+	hidden?: boolean;
+	options?: Record<string, any>;
+};
```

Comment:
What do you think of moving this type to _shared_ to not duplicate it between API and App?

Reply:
Yes 100%, I believe there's even a comment in one of the two saying I should 👀